thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/MultiThreadView.kt

```diff
@@ -44,71 +46,85 @@ import javax.swing.SortOrder
 import javax.swing.UIManager
 import kotlin.io.path.inputStream
 import kotlin.io.path.name
+import kotlin.io.path.nameWithoutExtension
 
 class MultiThreadView(
-    private val paths: List<Path>
+    private val paths: List<Path>,
 ) : ToolPanel() {
-    private val threadDumps = paths.mapNotNull { path -> ThreadDump.fromStream(path.inputStream()) }.toList()
+    private val threadDumps = paths.map { path ->
+        ThreadDump.fromStream(path.inputStream()) ?: throw ToolOpeningException("Failed to open $path as a thread dump")
+    }
+
+    private var visibleThreadDumps: List<ThreadDump> = emptyList()
```

Comment:
Initial value is emptyList so that things like the call to `toLifespanList()` can only happen from one place; there's lots of refactoring along those lines